src/main/drivers/barometer/barometer_dps310.c

```diff
@@ -0,0 +1,415 @@
+    if ((status & DPS310_MEAS_CFG_SENSOR_RDY) == 0) {
+        return false;
+    }
+
+    // 1. Read the pressure calibration coefficients (c00, c10, c20, c30, c01, c11, and c21) from the Calibration Coefficient register.
+    //   Note: The coefficients read from the coefficient register are 2's complement numbers.
+    uint8_t coef[18];
+    if (!busReadBuf(busDev, DPS310_REG_COEF, coef, sizeof(coef))) {
+        return false;
+    }
+
+    // 0x11 c0 [3:0] + 0x10 c0 [11:4]
+    baroState.calib.c0 = getTwosComplement(((uint32_t)coef[0] << 4) | (((uint32_t)coef[1] >> 4) & 0x0F), 12);
+
+    // 0x11 c1 [11:8] + 0x12 c1 [7:0]
+    baroState.calib.c1 = getTwosComplement((((uint32_t)coef[1] & 0x0F) << 8) | (uint32_t)coef[2], 12);
+
+    // 0x13 c00 [19:12] + 0x14 c00 [11:4] + 0x15 c00 [3:0]
+    baroState.calib.c00 = getTwosComplement(((uint32_t)coef[3] << 12) | ((uint32_t)coef[4] << 4) | (((uint32_t)coef[5] >> 4) & 0x0F), 20);
+
+    // 0x15 c10 [19:16] + 0x16 c10 [15:8] + 0x17 c10 [7:0]
+    baroState.calib.c10 = getTwosComplement((((uint32_t)coef[5] & 0x0F) << 16) | ((uint32_t)coef[6] << 8) | (uint32_t)coef[7], 20);
+
+    // 0x18 c01 [15:8] + 0x19 c01 [7:0]
+    baroState.calib.c01 = getTwosComplement(((uint32_t)coef[8] << 8) | (uint32_t)coef[9], 16);
+
+    // 0x1A c11 [15:8] + 0x1B c11 [7:0]
+    baroState.calib.c11 = getTwosComplement(((uint32_t)coef[8] << 8) | (uint32_t)coef[9], 16);
+
+    // 0x1C c20 [15:8] + 0x1D c20 [7:0]
+    baroState.calib.c20 = getTwosComplement(((uint32_t)coef[12] << 8) | (uint32_t)coef[13], 16);
+
+    // 0x1E c21 [15:8] + 0x1F c21 [7:0]
+    baroState.calib.c21 = getTwosComplement(((uint32_t)coef[14] << 8) | (uint32_t)coef[15], 16);
+
+    // 0x20 c30 [15:8] + 0x21 c30 [7:0]
+    baroState.calib.c30 = getTwosComplement(((uint32_t)coef[16] << 8) | (uint32_t)coef[17], 16);
+
+    // PRS_CFG: pressure measurement rate (32 Hz) and oversampling (16 time standard)
+    registerSetBits(busDev, DPS310_REG_PRS_CFG, DPS310_PRS_CFG_BIT_PM_RATE_32HZ | DPS310_PRS_CFG_BIT_PM_PRC_16);
+
+    // TMP_CFG: temperature measurement rate (32 Hz) and oversampling (16 times)
+    const uint8_t TMP_COEF_SRCE = registerRead(busDev, DPS310_REG_COEF_SRCE) & DPS310_COEF_SRCE_BIT_TMP_COEF_SRCE;
+    registerSetBits(busDev, DPS310_REG_TMP_CFG, DPS310_TMP_CFG_BIT_TMP_RATE_32HZ | DPS310_TMP_CFG_BIT_TMP_PRC_16 | TMP_COEF_SRCE);
+
+    // CFG_REG: set pressure and temperature result bit-shift (required when the oversampling rate is >8 times)
+    registerSetBits(busDev, DPS310_REG_CFG_REG, DPS310_CFG_REG_BIT_T_SHIFT | DPS310_CFG_REG_BIT_P_SHIFT);
+
+    // MEAS_CFG: Continuous pressure and temperature measurement
+    registerSetBits(busDev, DPS310_REG_MEAS_CFG, DPS310_MEAS_CFG_MEAS_CTRL_CONT);
+
+    return true;
+}
+
+static bool deviceReadMeasurement(baroDev_t *baro)
+{
+    // 1. Check if pressure is ready
+    bool pressure_ready = registerRead(&baro->busdev, DPS310_REG_MEAS_CFG) & DPS310_MEAS_CFG_PRS_RDY;
```

Comment:
Has this been tested in SPI mode? I suspect this might be prone to the same problem that was fixed for the MS5611 baro in #9735.

Reply:
Yes, the device is tested with SPI, too, and it works ;)